tornjak-frontend/src/redux/actions/types.ts

```diff
@@ -0,0 +1,119 @@
+import { Action } from "redux";
+import { selectors, workloadSelectors } from "data/data";
+import { 
+    AgentsListType, 
+    AgentsWorkLoadAttestorInfoType, 
+    ClustersListType, 
+    EntriesListType, 
+    ServerInfoType, 
+    TornjakServerInfoType 
+} from "components/types";
+
+// agents
+export const GLOBAL_AGENTS_LIST = 'GLOBAL_AGENTS_LIST';
+export const GLOBAL_AGENTS_WORKLOADATTESTOR_INFO = 'GLOBAL_AGENTS_WORKLOADATTESTOR_INFO'
+
+export interface AgentsReducerStateType {
+    globalAgentsList: AgentsListType[] | undefined,
+    globalAgentsWorkLoadAttestorInfo: AgentsWorkLoadAttestorInfoType[],
+}
+
+export interface AgentsListAction extends Action<typeof GLOBAL_AGENTS_LIST> {
+    payload: AgentsListType[];
+}
+export interface AgentWorkloadSelectorInfoAction extends Action<typeof GLOBAL_AGENTS_WORKLOADATTESTOR_INFO> {
```

Comment:
please, no  typeof

Reply:
This is because you need to infer the payload type because we have promise resolving middleware and the action type is explicitly defined.